stub-idp/src/integration-test/java/uk/gov/ida/apprule/steps/Cookies.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.ida.apprule.steps;
+
+import uk.gov.ida.stub.idp.cookies.CookieNames;
+import uk.gov.ida.stub.idp.cookies.HttpOnlyNewCookie;
+
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Cookies {
+    Map<String, NewCookie> cookies;
+
+    public Cookies() {
+        cookies = new HashMap<>();
+    }
+
+    public void extractCookies(Response response) {
+        response.getCookies().forEach((k,v) -> this.cookies.put(k,v));
```

Comment:
I decided not to swap the object references because then cookies might go missing.  PutAll would work better but thinking on it, I should probably do something to remove expired cookies.  Which I've done now (although for the integrations tests as-are it's admittedly redundant).